File: Leetcode 3801-3900/3810/3810.java
// Leetcode 3810: Minimum Operations to Reach Target Array
// https://leetcode.com/problems/minimum-operations-to-reach-target-array/
// Solved on 23rd of January, 2026
class Solution {
    /**
     * Calculates the minimum number of operations to reach the target array.
     * @param nums The initial array of integers.
     * @param target The target array of integers.
     * @return The minimum number of operations required.
     */
    public int minOperations(int[] nums, int[] target) {
        boolean[] seen = new boolean[100001];
        int operations = 0;
        for (int i = 0; i < nums.length; i++) {
            int current = nums[i];
            if (current != target[i]) {
                if (!seen[current]) {
                    seen[current] = true;
                    operations++;
                }
            }
        }
        return operations;
    }
}

File: Leetcode 3801-3900/3810/3810.py
# Leetcode 3810: Minimum Operations to Reach Target Array
# https://leetcode.com/problems/minimum-operations-to-reach-target-array/
# Solved on 23rd of January, 2026
class Solution:
    def minOperations(self, nums: list[int], target: list[int]) -> int:
        """
        Calculates the minimum number of operations to reach the target array.

        Args:
            nums (list[int]): The initial array of integers.
            target (list[int]): The target array of integers.

        Returns:
            int: The number of unique values in nums that need to be changed.
        """
        valuesToChange = set()

        for num, finalValue in zip(nums, target):
            if num != finalValue:
                valuesToChange.add(num)

        return len(valuesToChange)